### DISCUSSION TITLE
Which API?

### DISCUSSION THREAD

**Discussion by lucky-luke-98** [olafbombach/kg-lab-ws23-task3#8] [Ideas]
Which API for data import?

**Comment by christophe1998** [comment]
CORE
API:
https://core.ac.uk/services/api        (Core API is something entirely different! Do not confuse them)
Documentation: https://api.core.ac.uk/docs/v3

Advantages:
277 million papers
Multidisciplinary
Both a DB (36 million) and a metadata service
Free and no authentification for non-commercial use


Disadvantages:
Not free for commercial use
Cite core reference paper (Possibly optional and should be very easy)
https://core.ac.uk/about/research-outputs
Max 10000 results
API key necessary
contacting core necessary and possible display on the core website (https://core.ac.uk/faq#can-I-use-the-CORE-API-in-a-project)

Note:
I could not determine how to even use the API with the existing documentation.

**Comment by olafbombach** [comment]
Base: https://api.base-search.net/
good API documentation: https://www.base-search.net/about/download/base_interface.pdf
note: request for apikey or registered IP needed

**Comment by efeboz** [comment]
AMiner: https://www.aminer.org/
The documentations are not clear and information was hard to extract but here are some pages worth visiting:
-https://www.aminer.cn/restful_service
-http://doc.aminer.org/en/latest/
-https://api.aminer.org/ (web app?)
-https://www.aminer.org/gender/api (some functionality for gender information(?))
-https://github.com/AMinerOpen/prediction_api (prediction api with pre-trained model)

Good amount of data, overall hard to work with due to missing or hard to understand documentation

**Comment by lucky-luke-98** [comment]
# Crossref: https://www.crossref.org/
specs: multidisciplinary, ca. 110 million entries, free use without citation, no restrictions on commercial
## advantage: 
- offers an access free REST API https://api.crossref.org/swagger-ui/index.html
- always up to date since API gets updated every 20 mins / summary information every 24 hours
- even offers python packages so that it is easy to implement (depending on our usage) [crossrefapi](https://github.com/fabiobatalha/crossrefapi), [crossref-common](https://gitlab.com/crossref/crossref_commons_py) and [habanero](https://github.com/sckott/habanero)

In general, documentation seems good.

**Comment by lucky-luke-98** [comment]
# Europe PMC https://europepmc.org/
specs: focus on biomedical, ca. 30 million entries, free use
## advantages:
- offers multiple APIs (depending on what data we want to have)
- for us [OAI Service](https://europepmc.org/OaiService) should be sufficient. However, documentation seems bad and some links do not seem to work (also they mention, that they are currently not working on updates)
- their typical API [Article RESTful API](https://europepmc.org/RestfulWebService) seems well thought through
- *journal* is the property/key for the publishments

Are journal entries also wanted?

**Comment by christophe1998** [comment]
Worldcat:
Commercial API requires subscription https://www.oclc.org/developer/api/oclc-apis/worldcat-search-api.en.html
Integration into Python: https://pypi.org/project/bookops-worldcat/

Advantages:
Contains the contents of most libraries worldwide
Largest bibliographical database (150 million items)


Disadvantages:
Not commercial !
Commercial use only for participating libraries or similar institutions

**Comment by efeboz** [comment]
Google Scholar
https://scholar.google.com/

-Search engine that offers the largest amount of data.
-API usage remains a question since Google does not provide an official API and ready-to-use APIs are either outdated or commercial/subscription based. A solution would be to write a web scraper, but the feasibility of this solution is an open question.
